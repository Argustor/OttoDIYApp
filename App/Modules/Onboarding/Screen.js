import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  Carousel,
  Card
} from 'App/Components'

import { Images } from 'App/Themes'

export default class Screen extends Component {
  static propTypes = {
    onDone: PropTypes.func.isRequired
  }

  render () {
    const { onDone } = this.props

    return (
      <Container>
        <Carousel
          buttons={['Next', 'Next', 'Done']}
          onPress={[null, null, null]}
          onDone={onDone}>
          <Card
            image={Images.onboarding.one}
            title='Welcome to the Otto DIY App'
            text='Where your Otto comes alive'
            style={{titleViewText: {fontSize: 22}}}
          />
          <Card
            image={Images.onboarding.two}
            title='Otto awaits your commands'
            text='Otto DIY+ & Humanoid supported out of the box, more robots coming soon!'
            style={{titleViewText: {fontSize: 22}}}
          />
          <Card
            image={Images.onboarding.three}
            title='Lots more to come! ðŸŽ‰'
            text='Learn to code, customize your Otto and so much more...'
            style={{titleViewText: {fontSize: 22}}}
          />
        </Carousel>
      </Container>
    )
  }
}
