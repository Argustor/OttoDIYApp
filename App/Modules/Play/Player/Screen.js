import React, { Component } from 'react'
import { View, Image } from 'react-native'
import PropTypes from 'prop-types'

import {
  Container,
  ChamferImageButton,
  ArrowJoystick,
  Modal} from 'App/Components'

import { Images } from 'App/Themes'

import s from './Styles'
import { on } from 'cluster';

const speedButtonImages = {
  slow: Images.buttons.speed.slow,
  medium: Images.buttons.speed.medium,
  fast: Images.buttons.speed.fast
}

export default class Screen extends Component {
  static propTypes = {
    speed: PropTypes.string.isRequired,
    showNotConnectedModal: PropTypes.bool.isRequired,
    onUp: PropTypes.func,
    onDown: PropTypes.func,
    onLeft: PropTypes.func,
    onupdown: PropTypes.func,
    onmoonwalkright: PropTypes.func,
    onmoonwalkleft: PropTypes.func,
    onswing: PropTypes.func,
    oncrossright: PropTypes.func,
    oncrossleft: PropTypes.func,
    onflapfront: PropTypes.func,
    onflapback: PropTypes.func,
    ontiptoe: PropTypes.func,
    onbendright: PropTypes.func,
    onbendleft: PropTypes.func,
    onshakeright: PropTypes.func,
    onshakeleft: PropTypes.func,
    onjitter: PropTypes.func,
    onascend: PropTypes.func,
    onToggleSpeed: PropTypes.func,
    onHideNotConnectedModal: PropTypes.func.isRequired
  }

  render () {
    const {
      speed,
      showNotConnectedModal,
      onUp,
      onDown,
      onLeft,
      onRight,
      onupdown,
      onmoonwalkright,
      oncrossright,
      oncrossleft,
      onswing,
      onflapfront,
      onflapback,
      ontiptoe,
      onbendright,
      onbendleft,
      onshakeright,
      onshakeleft,
      onjitter,
      onascend,
      onToggleSpeed,
      onHideNotConnectedModal
    } = this.props
    return (
      <Container dark>
        <View style={s.header}>
          <Image source={Images.robots.plus} />
          <View style={s.headerCenter} />
          <Image source={Images.buttons.battery} />
          <Image source={Images.buttons.bluetooth} />
        </View>
        <View style={s.separator}>
          <View style={s.separatorDiagonal}>
            <View style={s.separatorDiagonalLeft} />
          </View>
          <View style={s.separatorCenter} />
          <View style={s.separatorDiagonal}>
            <View style={s.separatorDiagonalRight} />
          </View>
        </View>
        <View style={s.driveView}>
          <ChamferImageButton
            image={speedButtonImages[speed]}
            onPress={onToggleSpeed} />
          <View style={s.joystickView}>
            <ArrowJoystick
              onUp={onUp}
              onDown={onDown}
              onLeft={onLeft}
              onRight={onRight} />
          </View>
          <ChamferImageButton
            image={Images.buttons.help}
            onPress={() => {}} />
        </View>
        <View style={s.buttonsView}>
          <View style={s.buttonsRowView}>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.jitter}
                onPress={onjitter} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.swing}
                onPress={onswing} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.tiptoe}
                onPress={ontiptoe} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.moves}
                onPress={() => {}} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.updown}
                onPress={updown} />
            </View>
          </View>
          <View style={s.buttonsRowView}>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.flapback}
                onPress={flapback} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.moonwalkleft}
                onPress={moonwalkright} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.shakeleft}
                onPress={shakeleft} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.bendleft}
                onPress={bendleft} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.crossleft}
                onPress={crossleft} />
            </View>
          </View>
          <View style={s.buttonsRowView}>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.flapfront}
                onPress={flapfront} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.moonwalkright}
                onPress={moonwalkright} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.shakeright}
                onPress={shakeright} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.bendright}
                onPress={bendright} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.crossright}
                onPress={crossright} />
            </View>
          </View>
        </View>
        <View style={s.separator_flipped}>
          <View style={s.separatorDiagonal}>
            <View style={s.separatorDiagonalLeft} />
          </View>
          <View style={s.separatorCenter} />
          <View style={s.separatorDiagonal}>
            <View style={s.separatorDiagonalRight} />
          </View>
        </View>
        <View style={s.buttonsBottomView}>
          <View style={s.buttonsRowView}>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.updown}
                onPress={updown} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.beep}
                onPress={beep} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.gestures.happy}
                onPress={() => {}} />
            </View>
            <View style={s.buttonView}>
              <ChamferImageButton
                image={Images.buttons.ultrasound}
                onPress={() => {}} />
            </View>
          </View>
        </View>
        <Modal
          navigation={this.props.navigation}
          show={showNotConnectedModal}
          onHidePress={onHideNotConnectedModal}
          template='NotConnected' />
      </Container>
    )
  }
}
