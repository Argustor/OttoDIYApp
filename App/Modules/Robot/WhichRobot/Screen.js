import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Container, List, Footer, Card, CardListItem } from 'App/Components'
import { Images } from 'App/Themes'

export default class WhichRobotScreen extends Component {
  static propTypes = {
    onPress: PropTypes.func.isRequired
  }

  render () {
    const { onPress } = this.props
    return (
      <Container>
        <List linedRows scrollable>
          <CardListItem
            image={Images.robots.plus}
            title='Otto DIY'
            text='The open source robot that you can build yourself'
            onPress={() => { onPress('otto') }} />
        </List>
        <Footer>
          <Card theme='light' text='Choose the robot you want to play with' />
        </Footer>
      </Container>
    )
  }
}
