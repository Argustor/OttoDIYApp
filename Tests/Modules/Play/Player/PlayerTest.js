import React from 'react'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Screen from 'App/Modules/Play/Player/Screen'

describe('Player', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Screen
        navigation={{navigate: () => {}}}
        speed='medium'
        showNotConnectedModal={false}
        onUp={() => {}}
        onDown={() => {}}
        onLeft={() => {}}
        onRight={() => {}}
        onToggleSpeed={() => {}}
        onHideNotConnectedModal={() => {}} />)
  })

  it('Component renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
