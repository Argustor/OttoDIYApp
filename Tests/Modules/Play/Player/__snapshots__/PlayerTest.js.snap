// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Component renders correctly 1`] = `
<Container
  dark={true}
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "margin": 16,
        "marginBottom": 0,
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    />
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
    />
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "margin": 16,
      }
    }
  >
    <View
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    >
      <View
        style={
          Object {
            "borderTopColor": "white",
            "borderTopWidth": 2,
            "height": 20,
            "marginLeft": -14,
            "transform": Array [
              Object {
                "rotate": "135deg",
              },
            ],
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "borderTopColor": "white",
          "borderTopWidth": 2,
          "flex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    >
      <View
        style={
          Object {
            "borderTopColor": "white",
            "borderTopWidth": 2,
            "height": 20,
            "marginRight": -14,
            "transform": Array [
              Object {
                "rotate": "-135deg",
              },
            ],
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <ChamferImageButton
      image={
        Object {
          "process": [Function],
        }
      }
      onPress={[Function]}
    />
    <View
      style={
        Object {
          "marginHorizontal": 16,
        }
      }
    >
      <ArrowJoystick
        onDown={[Function]}
        onLeft={[Function]}
        onRight={[Function]}
        onUp={[Function]}
      />
    </View>
    <ChamferImageButton
      image={
        Object {
          "process": [Function],
        }
      }
      onPress={[Function]}
    />
  </View>
  <View
    style={
      Object {
        "marginVertical": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "margin": 16,
        "transform": Array [
          Object {
            "rotate": "180deg",
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    >
      <View
        style={
          Object {
            "borderTopColor": "white",
            "borderTopWidth": 2,
            "height": 20,
            "marginLeft": -14,
            "transform": Array [
              Object {
                "rotate": "135deg",
              },
            ],
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "borderTopColor": "white",
          "borderTopWidth": 2,
          "flex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    >
      <View
        style={
          Object {
            "borderTopColor": "white",
            "borderTopWidth": 2,
            "height": 20,
            "marginRight": -14,
            "transform": Array [
              Object {
                "rotate": "-135deg",
              },
            ],
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "margin": 8,
          }
        }
      >
        <ChamferImageButton
          image={
            Object {
              "process": [Function],
            }
          }
          onPress={[Function]}
        />
      </View>
    </View>
  </View>
  <Modal
    navigation={
      Object {
        "navigate": [Function],
      }
    }
    onHidePress={[Function]}
    show={false}
    template="NotConnected"
  />
</Container>
`;
